packages/vite/src/middleware/CookieJar.test.ts

```diff
@@ -0,0 +1,83 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+/* to keep the tests a little cleaner by using ! */
+
+import { describe, expect, test } from 'vitest'
+
+import { CookieJar } from './CookieJar'
+
+describe('CookieJar', () => {
+  // Grabbed from github
+  const cookieJar = new CookieJar(
+    'color_mode=%7B%22color_mode%22%3A%22light%22%2C%22light_theme%22%3A%7B%22name%22%3A%22light%22%2C%22color_mode%22%3A%22light%22%7D%2C%22dark_theme%22%3A%7B%22name%22%3A%22dark_dimmed%22%2C%22color_mode%22%3A%22dark%22%7D%7D; preferred_color_mode=dark; tz=Asia%2FBangkok'
+  )
+
+  test('instatitates cookie jar from a cookie string', () => {
+    expect(cookieJar.get('color_mode')).toStrictEqual({
+      value: JSON.stringify({
+        color_mode: 'light',
+        light_theme: { name: 'light', color_mode: 'light' },
+        dark_theme: { name: 'dark_dimmed', color_mode: 'dark' },
+      }),
+    })
+
+    expect(cookieJar.get('preferred_color_mode')).toStrictEqual({
+      value: 'dark',
+    })
+
+    expect(cookieJar.get('tz')).toStrictEqual({
+      value: 'Asia/Bangkok',
+    })
+  })
+
+  describe('Helper methods like JS Map', () => {
+    test('has', () => {
+      expect(cookieJar.has('color_mode')).toBe(true)
+      expect(cookieJar.has('bazinga')).toBe(false)
+    })
+
+    test('size', () => {
+      expect(cookieJar.size).toBe(3)
+    })
+
+    test('entries', () => {
+      const iterator = cookieJar.entries()
+
+      expect(iterator.next().done).toBe(false)
+      expect(iterator.next().done).toBe(false)
+
+      const finalItem = iterator.next()
+      expect(finalItem.done).toBe(false)
+      expect(finalItem.value).toStrictEqual(['tz', { value: 'Asia/Bangkok' }])
+
+      expect(iterator.next().done).toBe(true)
+    })
+
+    // @MARK: API convention worth discussing!
+    // Delete is a little special, it doesn't actually delete the cookie
+    // but sets it to expire and sets an empty value
```

Comment:
The alternative: delete the cookie from the map (like `clear('myCookie')`)

Its a question of intention, because I think in the context of middleware, you are looking to unset a cookie. e.g. If the browser had a `Session` cookie set, you want the browser to remove it (the serve can only send instructions, not actually modify the cookies) - it has to expire it. If you just deleted it from the map it would have no effect on the cookeis in your browser

Reply:
Maybe we can find another word to use. Like `invalidate`, `revoke`, `clear`, `unset`?

Also had to ask ChatGPT 😁 

> There are several synonyms for "delete," and the choice of word may depend on the context. Here are some alternatives:
> 
> 1. Remove
> 2. Erase
> 3. Expunge
> 4. Eliminate
> 5. Omit
> 6. Purge
> 7. Cancel
> 8. Wipe
> 9. Discard
> 10. Trash
> 11. Abolish
> 12. Exterminate
> 13. Annihilate
> 14. Obliterate
> 15. Uninstall (commonly used in the context of computer software)

Reply:
Hmmmm yes - so thw two constraints we have are:
1. The action I described above. `unset` or `revoke` seems quite clear for me, and we wouldn't have the delete method.
2. I'm also partially mirroring the methods available on `Map`. This isn't a requirement, but more a case of it feeling familiar/ergonomic.

Reply:
I've renamed it to unset, to make it clear its not the same as the Map.delete :)